tests/unittest/_torch/visual_gen/test_offloading.py

```
@@ -0,0 +1,267 @@
+    group = _ToyModule(weight_value=1.0, bias_value=10.0)
+    manager = ModuleOffloadManager(
+        groups={"group": group},
+        device="cpu",
+        pin_memory=False,
+    )
+    layout = manager._collect_group_layout("group", group, 0)
+    manager.cpu_storage = torch.empty(0, dtype=torch.uint8, device="cpu")
+
+    with pytest.raises(
+        RuntimeError,
+        match="Failed to copy offload tensor 'group.weight'",
```

Ruff (0.15.13)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)